### PR TITLE
ID into PDA without PDA in your inventory and a few other similar things

### PR DESCRIPTION
**Pull request by kingofkosmos**
Enables putting an ID into a PDA when the PDA is not in your inventory, e.g. the floor. This was probably an unreported bug because [this comment says it should work](https://github.com/tgstation/-tg-station/blob/d454eae1b1f84c2dac30bc29ee1c0917572c1897/code/game/objects/items/devices/PDA/PDA.dm#L890) but it doesn't.

Also fixes another unreported bug of altclicking a PDA on the floor to eject an ID, it always gave the message "You cannot do that while restrained!".

Also removes the same must-be-in-hands-requirements from attaching seclite to taser, and suppressor to gun, since every other item-combining action in the game can happen while the item is lying on the floor.

### CODE REVIEW

**Comment by Aranclanos** [comment]
the can_use() proc is horrible and old, swap it to canusetopic
i think you're enabling using the PDA href links even if you don't have the pda yourself

**Comment by Zilenan91** [comment]
I was wondering why I had to have my PDA in my hand to put my ID into it.

**Comment by kingofkosmos** [comment]
>i think you're enabling using the PDA href links even if you don't have the pda yourself

Crap, you're right I tested it. Switching it to canusetopic.

**Comment by kingofkosmos** [comment]
It's done. Also fixed another unreported bug of altclicking a PDA on the floor to eject an ID, it always gave the message "You cannot do that while restrained!".

File: code/game/objects/items/devices/PDA/PDA.dm [hunk 1]
```diff
@@ -794,13 +785,15 @@ var/global/list/obj/item/device/pda/PDAs = list()
...
 
-	if(can_use(usr))
+	if(usr.canUseTopic(src))
 		if(id)
 			remove_id()
 		else
 			usr << "<span class='warning'>This PDA does not have an ID in it!</span>"
-	else
+	if (usr.restrained())
 		usr << "<span class='warning'>You cannot do that while restrained!</span>"
```
Comment (Aranclanos) [inline code/game/objects/items/devices/PDA/PDA.dm]: this is unneeded

**Comment by Aranclanos** [comment]
the canusetopic check in the attackby doesn't seem needed, and you have to review all rejection messages (that are all copypaste, mostly a "you can't do this while restrained"). Remove all checks and messages that aren't needed
once that's done it should be good to go

File: code/game/objects/items/devices/PDA/PDA.dm (hunk 1 above)
Comment (kingofkosmos) [inline code/game/objects/items/devices/PDA/PDA.dm]: The whole message is unneeded? CanUseTopic doesn't give the message by itself.

Comment (Aranclanos) [inline code/game/objects/items/devices/PDA/PDA.dm]: yeah it's okay don't worry, we have other ways to tell the player that they are restrained or stunned or whatever

Comment (kingofkosmos) [inline code/game/objects/items/devices/PDA/PDA.dm]: OK will do.

**Comment by kingofkosmos** [comment]
Done. Also removed the same must-be-in-hands-requirements from attaching seclite to taser, and suppressor to gun.

**Comment by Bawhoppen** [comment]
does it work for putting pens and carts into PDA too now (if it didn't before)

**Comment by kingofkosmos** [comment]
Yup, pens and cartridges seem to work!

**Pull request merged**